tests/testthat/test-combinevariablesetsasbinary.R

```diff
@@ -0,0 +1,111 @@
+library(testthat)
+library(verbs)
+context("Combine variable sets as binary")
+
+
+data(phone, package = "flipExampleData")
+data(colas, package = "flipExampleData")
+
+
+phone.names <- c("AAPT", "New Tel", "One-tel", "Optus", "Orange", "Telstra", "Virgin", "Vodafone", "Other 1", "Other 2", "Don't know")
+unaided <- data.frame(phone[, paste0("Q5_", 1:11)])
+colnames(unaided) <- phone.names
+unaided[1, ] <- NA
+unaided = (unaided == "Yes") * 1
+attr(unaided, "questiontype") <- "PickAny"
+attr(unaided, "dataset") <- "phone"
+
+aided = data.frame(phone[, paste0("Q6_", 1:11)])
+colnames(aided) <- phone.names
+aided = (aided == "Yes") * 1
+attr(aided, "questiontype") <- "PickAny"
+attr(aided, "dataset") <- "phone"
+
+Q4 = as.data.frame(colas[, c("q4a", "q4b", "q4c", "q4d", "q4e", "q4f")])
+attr(Q4, "questiontype") <- "PickOneMulti"
+attr(Q4, "dataset") <- "colas"
+
+Q4.small = as.data.frame(colas[, c("q4a", "q4b", "q4c", "q4d")])
+attr(Q4.small, "questiontype") <- "PickOneMulti"
+attr(Q4.small, "dataset") <- "colas"
+
+Q4.pepsi.light = colas[, "q4e"]
+attr(Q4.pepsi.light, "questiontype") <- "PickOne"
+attr(Q4.pepsi.light, "dataset") <- "colas"
+
+Q4.pepsi.max = colas[, "q4f"]
+attr(Q4.pepsi.max, "questiontype") <- "PickOne"
+attr(Q4.pepsi.max, "dataset") <- "colas"
+
+Q4.binary = CombineVariableSetsAsBinary(list(Q4))
+attr(Q4.binary, "questiontype") <- "PickAny"
+attr(Q4.binary, "dataset") <- "colas"
+
+Q4.binary.small = CombineVariableSetsAsBinary(list(Q4))
+attr(Q4.binary.small, "questiontype") <- "PickAny"
+attr(Q4.binary.small, "dataset") <- "colas"
+
+test_that("Single PickOne", {
+
+	asnumeric = flipTransformations::AsNumeric(Q4.pepsi.light, binary = TRUE)
+	colnames(asnumeric) = levels(Q4.pepsi.light)
+	asnumeric = asnumeric == 1
+
+	expect_equal(CombineVariableSetsAsBinary(list(Q4.pepsi.light)), asnumeric)
+
+})
+
+test_that("Two PickOnes", {
+
+	pepsi.light.numeric = flipTransformations::AsNumeric(Q4.pepsi.light, binary = TRUE)
+	colnames(pepsi.light.numeric) = levels(Q4.pepsi.light)
+	pepsi.max.numeric = flipTransformations::AsNumeric(Q4.pepsi.max, binary = TRUE)
+	colnames(pepsi.max.numeric) = levels(Q4.pepsi.max)
+	input.args = list(pepsi.light.numeric, pepsi.max.numeric)
+	input.args[["match.elements"]] <- "Yes"
+    input.args[["elements.to.count"]] <- list(numeric = 1, categorical = NULL)
+    input.args[["ignore.missing"]] <- TRUE
+	pepsi.light.or.max = do.call(AnyOf, input.args)
+
+	expect_equal(CombineVariableSetsAsBinary(list(Q4.pepsi.light, Q4.pepsi.max)), pepsi.light.or.max)
+})
+
+test_that("Many PickOnes are equivalent to a PickOneMulti", {
+
+	expect_equal(CombineVariableSetsAsBinary(list(Q4)), CombineVariableSetsAsBinary(list(colas[, "q4a"], colas[, "q4b"], colas[, "q4c"], colas[, "q4d"], colas[, "q4e"], colas[, "q4f"])))
```

Comment:
This could be simplified to `CombineVariableSetsAsBinary(Q4)` and similar comma separated argument for the second call if the generic `...` implementation is used.